test/hotspot/jtreg/serviceability/sa/UniqueVtableTest.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * @run main/othervm --add-opens=jdk.hotspot.agent/sun.jvm.hotspot.types.basic=ALL-UNNAMED UniqueVtableTest
+ */
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import sun.jvm.hotspot.HotSpotAgent;
+import sun.jvm.hotspot.debugger.Address;
+import sun.jvm.hotspot.types.Type;
+import sun.jvm.hotspot.types.basic.BasicTypeDataBase;
+
+import jdk.test.lib.apps.LingeredApp;
+import jdk.test.lib.SA.SATestUtils;
+
+
+public class UniqueVtableTest {
+
+    private HotSpotAgent agent;
+    private MethodHandle vtblForType;
+
+    public UniqueVtableTest() {
+    }
+
+    private static String type2String(Type t) {
+        return t + " (extends " + t.getSuperclass() + ")";
+    }
+
+    private static void log(Object o) {
+        System.out.println(o);
+    }
+
+    private void attach(long pid) throws Throwable {
+        agent = new HotSpotAgent();
+        log("Attaching to process ID " + pid + "...");
+        agent.attach((int) pid);
+        log("Attached successfully.");
+
+        // agent.getTypeDataBase() returns HotSpotTypeDataBase (extends BasicTypeDataBase)
+        // We need a method from BasicTypeDataBase
+        //    Address vtblForType(Type type);
+
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        MethodHandles.Lookup classLookup = MethodHandles.privateLookupIn(BasicTypeDataBase.class, lookup);
+        vtblForType = classLookup.findVirtual(BasicTypeDataBase.class, "vtblForType",
+                 MethodType.methodType(Address.class, Type.class));
+    }
+
+    private void detach() {
+        if (agent != null) {
+            agent.detach();
+        }
+    }
+
+    private void runTest() throws Throwable {
+        Map<Address, List<Type>> types = new HashMap<>();
+        Iterator<Type> it = agent.getTypeDataBase().getTypes();
+        int dupFound = 0;
+        while (it.hasNext()) {
+            Type t = it.next();
+            Address vtable = (Address) vtblForType.invoke(agent.getTypeDataBase(), t);
+
+            if (vtable != null) {
+                List<Type> typeList = types.get(vtable);
+                if (typeList == null) {
+                    types.put(vtable, new ArrayList<>(List.of(t)));
+                } else {
+                    // duplicate found
+                    dupFound++;
+                    typeList.add(t);
+                }
+            }
+
+            if (vtable == null && t.getSuperclass() != null) {
+               log("WARNING: vtable is null for " + type2String(t)
+                  + ", CInt: " + t.isCIntegerType()
+                  + ", CStr: " + t.isCStringType()
+                  + ", JPrimitive: " + t.isJavaPrimitiveType()
+                  + ", Oop: " + t.isOopType()
+                  + ", Ptr: " + t.isPointerType());
```

Comment:
It appears that these always print "false". Are they worth having?

Reply:
It reports all available info about the Type. As these are "suspicious" types I think it make sense to provide more info

Reply:
removed this property as by the updated condition logged classes have all "false"